src/Microsoft.OData.Core/UriParser/Parsers/FunctionCallParser.cs

```diff
@@ -97,14 +97,14 @@ public bool TryParseIdentifierAsFunction(QueryToken parent, out QueryToken resul
             else
             {
                 Debug.Assert(this.Lexer.CurrentToken.Kind == ExpressionTokenKind.Identifier, "Only identifier tokens can be treated as function calls.");
-                functionName = this.Lexer.CurrentToken.Text;
+                functionName = this.Lexer.CurrentToken.Span;
                 this.Lexer.NextToken();
             }
 
             FunctionParameterToken[] arguments = this.ParseArgumentListOrEntityKeyList(() => lexer.RestorePosition(position));
             if (arguments != null)
             {
-                result = new FunctionCallToken(functionName, arguments, parent);
+                result = new FunctionCallToken(functionName.ToString(), arguments, parent);

```

Comment:
Should we consider making the `FunctionCallToken` and other query tokens accept `ReadOnlySpan<char>` or `ReadOnlyMemory<T>` instead of `string`? I think that would reduce allocations further. I'm going off the assumption that these tokens are the generated from a lexer, and therefore always tied to some input query string expression, and so the tokens can also point into segments of the original string instead of being requiring allocations. The downside is the source string would remain in memory for the lifetime of the tokens.

I don't know if we expect users to instantiate these tokens without going through a lexer, but since the constructors are public, I guess that possible. But we can still store span/memory without a breaking change I think, and the string properties could be computed from the span/memory on demand then cached. If we have a clean separation between the syntax and semantic parse phases, then the lexer and lexical tokens could be short-lived most of the time, maybe we could be more aggressive about using spans and even ref structs.